services/storage/src/simcore_service_storage/simcore_s3_dsm.py

```diff
@@ -106,17 +107,55 @@ async def list_datasets(self, user_id: UserID) -> list[DatasetMetaData]:
             ]
 
     async def list_files_in_dataset(
-        self, user_id: UserID, dataset_id: str
+        self, user_id: UserID, dataset_id: str, expand_dirs: bool
     ) -> list[FileMetaData]:
         data: list[FileMetaData] = await self.list_files(
-            user_id, uuid_filter=dataset_id + "/"
+            user_id, expand_dirs, uuid_filter=ensure_ends_with(dataset_id, "/")
         )
         return data
 
+    async def _expand_directory(self, fmd: FileMetaDataAtDB) -> list[FileMetaData]:
+        """
+        Scans S3 backend and returns a list S3MetaData entries which get mapped
+        to FileMetaData entry.
+        """
+        files_in_folder: list[S3MetaData] = await get_s3_client(self.app).list_files(
+            self.simcore_bucket_name,
+            prefix=ensure_ends_with(fmd.file_id, "/"),
+        )
+        result: list[FileMetaData] = [
+            convert_db_to_model(
+                FileMetaDataAtDB(
+                    location_id=fmd.location_id,
+                    location=fmd.location,
+                    bucket_name=fmd.bucket_name,
+                    object_name=x.file_id,
+                    user_id=fmd.user_id,
+                    # NOTE `created_at` is inherited from the directory,
```

Comment:
interesting. does S3 not provide any date on file creation???
I think you should have that entry. it's called Last-Modified.
see [https://docs.aws.amazon.com/AmazonS3/latest/userguide/UsingMetadata.html](https://docs.aws.amazon.com/AmazonS3/latest/userguide/UsingMetadata.html)

Reply:
even thoiugh there is`last_modified` below coming from S3.since we do not get a `created_at` from the file. it's more convenient to have it the same as the `last_modified`. Otherwise when adding files to the directory in the future this field would become irrelevant and confusing for the users. This way it behaves the same.